includes/api/class-wc-admin-rest-onboarding-profile-controller.php

```diff
@@ -0,0 +1,300 @@
+		 * collection request.
+		 *
+		 * @param array           $args    Key value array of query var to query value.
+		 * @param WP_REST_Request $request The request used.
+		 */
+		$args = apply_filters( 'woocommerce_rest_onboarding_profile_object_query', $args, $request );
+
+		return $args;
+	}
+
+
+	/**
+	 * Prepare the data object for response.
+	 *
+	 * @param object          $item Data object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$data     = $this->add_additional_fields_to_object( $item, $request );
+		$data     = $this->filter_response_by_context( $data, 'view' );
+		$response = rest_ensure_response( $data );
+
+		/**
+		 * Filter the list returned from the API.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param array            $item     The original item.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_onboarding_profile', $response, $item, $request );
+	}
+
+	/**
+	 * Get onboarding profile properties.
+	 *
+	 * @return array
+	 */
+	public static function get_profile_properties() {
+		$properties = array(
+			'account_type'   => array(
+				'type'        => 'string',
+				'description' => __( 'Account type used for Jetpack.', 'woocommerce-admin' ),
+				'context'     => array( 'view' ),
+				'readonly'    => true,
+			),
+			'industry'       => array(
+				'type'        => 'string',
+				'description' => __( 'Industry.', 'woocommerce-admin' ),
+				'context'     => array( 'view' ),
+				'readonly'    => true,
+			),
+			'product_types'  => array(
+				'type'        => 'array',
+				'description' => __( 'Types of products sold.', 'woocommerce-admin' ),
+				'context'     => array( 'view' ),
+				'readonly'    => true,
+				'items'       => array(
+					'type' => 'string',
+				),
+			),
+			'product_count'  => array(
+				'type'        => 'integer',
+				'description' => __( 'Number of products to be added.', 'woocommerce-admin' ),
+				'context'     => array( 'view' ),
+				'readonly'    => true,
+			),
+			'selling_venues' => array(
+				'type'        => 'array',
+				'description' => __( 'Other places the store is selling products.', 'woocommerce-admin' ),
+				'context'     => array( 'view' ),
+				'readonly'    => true,
+				'items'       => array(
+					'type' => 'string',
+				),
+			),
+			'other_platform' => array(
+				'type'        => 'string',
+				'description' => __( 'Name of other platform used to sell.', 'woocommerce-admin' ),
+				'context'     => array( 'view' ),
+				'readonly'    => true,
+			),
+			'theme'          => array(
+				'type'        => 'string',
+				'description' => __( 'Selected store theme.', 'woocommerce-admin' ),
+				'context'     => array( 'view' ),
+				'readonly'    => true,
+			),
+			'purchase'       => array(
```

Comment:
In addition to the `purchase` bool, we should probably have another boolean indicating if they skipped the entire profiler.

Reply:
Added in https://github.com/woocommerce/woocommerce-admin/pull/2217/commits/c2252a3738966745db716e202276ff173603eda6